crates/transpiler/src/passes/inverse_cancellation.rs

```diff
@@ -156,9 +156,113 @@ fn run_on_inverse_pairs(
     Ok(())
 }
 
+static SELF_INVERSE_GATES_FOR_CANCELLATION: [StandardGate; 9] = [
+    StandardGate::CX,
+    StandardGate::ECR,
+    StandardGate::CY,
+    StandardGate::CZ,
+    StandardGate::X,
+    StandardGate::Y,
+    StandardGate::Z,
+    StandardGate::H,
+    StandardGate::Swap,
+];
+
+static INVERSE_PAIRS_FOR_CANCELLATION: [[StandardGate; 2]; 3] = [
+    [StandardGate::T, StandardGate::Tdg],
+    [StandardGate::S, StandardGate::Sdg],
+    [StandardGate::SX, StandardGate::SXdg],
+];
+
+#[pyfunction]
+pub fn run_inverse_cancellation_standard_gates(dag: &mut DAGCircuit) {
+    if !SELF_INVERSE_GATES_FOR_CANCELLATION
+        .iter()
+        .chain(INVERSE_PAIRS_FOR_CANCELLATION.iter().flatten())
+        .any(|gate| dag.get_op_counts().contains_key(gate.name()))
+    {
+        return;
+    }
+
+    for self_inv_gate in SELF_INVERSE_GATES_FOR_CANCELLATION {
+        let filter = |inst: &PackedInstruction| -> bool {
+            match inst.op.view() {
+                OperationRef::StandardGate(gate) => gate == self_inv_gate,
+                _ => false,
+            }
+        };
+        let run_nodes: Vec<_> = dag.collect_1q_runs_by(filter).collect();
+        for gate_cancel_run in run_nodes {
+            let mut partitions: Vec<Vec<NodeIndex>> = Vec::new();
+            let mut chunk: Vec<NodeIndex> = Vec::new();
+            let max_index = gate_cancel_run.len() - 1;
+            for (i, cancel_gate) in gate_cancel_run.iter().enumerate() {
+                let node = &dag[*cancel_gate];
+                let NodeType::Operation(inst) = node else {
+                    unreachable!("Not an op node");
+                };
```

Comment:
I think that you can use ``NodeType::unwrap_operation()`` here iirc.

```suggestion
                let inst = node.unwrap_operation();
```

Reply:
I was just following jakelishman's lead https://github.com/Qiskit/qiskit/pull/14317#discussion_r2104637979